README.md

```diff
@@ -110,6 +207,27 @@ runResource finish = interpretH $ \case
         runIt = finish .@ runResource
 
     sendM $ X.bracket (runIt a) (runIt . d) (runIt . u)
+
+data CustomException = ThisException | ThatException deriving Show
+instance X.Exception CustomException
+
+program :: Members '[Resource, Teletype, Error CustomException] r => Sem r ()
+program = catch work $ \e -> writeTTY ("Caught " ++ show e)
+  where work = bracket (readTTY) (const $ writeTTY "exiting bracket") $ \input -> do
+          writeTTY "entering bracket"
+          when (input == "explode") $ throw ThisException
+          writeTTY input
+          when (input == "weird stuff") $ throw ThatException
+          writeTTY "no exceptions"
+
+boringProgram :: IO (Either CustomException ([String], ()))
+boringProgram = (runM .@ runResource) . runError @CustomException . runTeletypeBoring ["words"] $ program
+
+boringExplosion :: IO (Either CustomException ([String], ()))
+boringExplosion = (runM .@ runResource) . runError @CustomException . runTeletypeBoring ["explode"] $ program
+
+ioProgram :: IO (Either CustomException ())
+ioProgram = (runM .@ runResource .@@ runErrorInIO @CustomException) . runTeletypeIO $ program
```

Comment:
What do you think about moving these into a `docs/` page, with more prose/explanation power, + a link from the readme?